uniswap_swap_detection/src/agent.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  Finding,
+  TransactionEvent,
+  FindingSeverity,
+  FindingType,
+  ethers,
+  getEthersProvider,
+} from "forta-agent";
+
+import { SWAP_EVENT } from "./constant";
+import { poolABI } from "./abi";
+
+import { uniswapPoolAddress } from "./utils";
+
+interface HandleArgs {
+  PROVIDER: ethers.providers.JsonRpcProvider;
+}
+const PROVIDER = getEthersProvider();
+const block = 20;
```

Comment:
If you were attempting to use this `block` as the `blockTag` when using `ethers`, it is not a good idea to use a hardcoded value. This is because the value of a variable can change. So for example, if you are querying the value of a variable on a smart contract at block `20`, but the variable changed in block `21`, you would not get an updated value because you would be always querying at the same (i.e. stale) block.

That's why you should use `TransactionEvent.blockNumber`, because it'll be up to date until the current block number.